src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -345,9 +366,11 @@ private bool HasHome(BoundExpression expression, bool needWriteable)
                         ((BoundParameter)expression).ParameterSymbol.RefKind != RefKind.RefReadOnly;
 
                 case BoundKind.Local:
-                    // locals have home unless they are byval stack locals
+                    // locals have home unless they are byval stack locals or ref-readonly
+                    // locals in a mutating call
                     var local = ((BoundLocal)expression).LocalSymbol;
-                    return !IsStackLocal(local) || local.RefKind != RefKind.None;
+                    return !((IsStackLocal(local) && local.RefKind == RefKind.None) || 
+                        (needWriteable && local.RefKind == RefKind.RefReadOnly));
```

Comment:
Is needWritable protecting the local or the value?

Reply:
The local. It's specifying that you must be able to write to the local pointed to by the address.